onnxruntime/src/runner.rs

```diff
@@ -0,0 +1,244 @@
+use std::ffi::CString;
+use std::fmt::Debug;
+use std::ops::{Index, IndexMut};
+
+use ndarray::{Array, Dimension, IxDyn};
+
+use onnxruntime_sys as sys;
+
+use crate::error::{status_to_result, OrtError};
+use crate::memory::MemoryInfo;
+use crate::session::{Output, Session};
+use crate::tensor::OrtTensor;
+use crate::{g_ort, Result, TypeToTensorElementDataType};
+
+pub trait Element: 'static + Clone + Debug + TypeToTensorElementDataType + Default {}
+
+impl<T: 'static + Clone + Debug + TypeToTensorElementDataType + Default> Element for T {}
+
+fn names_to_ptrs(names: impl Iterator<Item = String>) -> Vec<*const i8> {
+    names
+        .map(|name| CString::new(name.clone()).unwrap().into_raw() as *const _)
+        .collect()
+}
+
+fn compute_output_shapes<TIn, DIn: Dimension>(
+    input_arrays: &[Array<TIn, DIn>],
+    outputs: &[Output],
+) -> Vec<Vec<usize>> {
+    outputs
+        .iter()
+        .enumerate()
+        .map(|(idx, output)| {
+            output
+                .dimensions
+                .iter()
+                .enumerate()
+                .map(|(jdx, dim)| match dim {
+                    None => input_arrays[idx].shape()[jdx],
+                    Some(d) => *d as usize,
+                })
```

Comment:
This makes a bad assumption about the output shape corresponding to the input shape (see https://github.com/nbigaouette/onnxruntime-rs/pull/56).

Is there an ORT API that can be used to get the output shape before execution? If not, can we allow users to specify the output shape?